spec/templates/markup_processor_integrations/markdown_spec.rb

```diff
@@ -72,26 +72,17 @@
 
     include_examples 'shared examples for markdown processors'
 
-
-    it 'generates anchor tags for level 2 header' do
-      expect(rendered_document).to include('<h2 id="example-code-listings">Example code listings</h2>')
-    end
-
     it 'does not create line break via backslash' do
       expect(rendered_document).to include("commonmark line break with\\\na backslash")
     end
   end
 
-  describe 'CommonMarker', if:  RUBY_VERSION >= '2.3' do
+  describe 'Commonmarker', if: RUBY_VERSION >= '2.3' do
     let(:markup) { :markdown }
     let(:markup_provider) { :commonmarker }
 
     include_examples 'shared examples for markdown processors'
 
-    it 'generates level 2 header without id' do
```

Comment:
This should remain but should be scoped to CommonMarker

Reply:
Why? Does it have an impact?

* > Testing providers’ implementation is outside of the scope of YARD.
* > Remove tests for Markdown header IDs or the lack of, considering they may improve with newer provider versions and that YARD shouldn’t constraint their growth

Reply:
> Testing providers’ implementation is outside of the scope of YARD.

Not when we're explicitly testing system integration. The fact that this behavior changes is exactly why we have tests. An id-less h2 heading actually has implications for downstream YARD code, since we have special code paths for heading id attributes.

Actually, expanding this comment outward, it seems like this change removes all integration tests for CommonMarker, which means we're now only testing Commonmarker 1.0. This is also insufficient. We should keep the old tests for CommonMarker and add the new ones for this 1.0 version, not replace.

tl;dr we need to test both CommonMarker / Commonmarker implementations. Not just one. You can remove the h2 test for Commonmarker 1.0 if you want, but both libraries should be integration tested.

Reply:
> Remove tests for Markdown header IDs or the lack of, considering they may improve with newer provider versions and that YARD shouldn’t constraint their growth

YARD relies on this IDs though, that's the issue, and that's why we test.